src/python/pants/core/goals/package.py

```diff
@@ -178,5 +184,30 @@ async def package_asset(workspace: Workspace, dist_dir: DistDir) -> Package:
     return Package(exit_code=0)
 
 
+def transitive_targets_without_traversing_packages_request(
+    *,
+    roots: Iterable[Address],
+    union_membership: UnionMembership,
+    always_traverse_roots: bool = True,  # traverse roots even if they are package targets
+) -> TransitiveTargetsRequest:
+    package_field_set_types = union_membership.get(PackageFieldSet)
+
+    def should_traverse_deps(tgt: Target, fld: Field) -> bool:
+        if not isinstance(fld, Dependencies):  # SpecialCasedDependencies
```

Comment:
Should we swap this and make it `if isinstance(field, SpecialCasedDependencies)`?